inflate.c

```diff
@@ -887,7 +887,7 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
             }
             unsigned char *next_out = state->window + state->wsize + state->wnext;
             if (copy <= state->offset) {
-                functable.chunkcopy_safe(next_out, next_out - state->offset, copy, put + buf_left);
+                chunkcopy_safe(next_out, next_out - state->offset, copy, put + buf_left);
```

Comment:
Return value is ignored here.

Reply:
Ahh that's a good point but it was always ignored, as you can see in the diff.  Am I supposed to assign that to next_out?

Reply:
From the looks of it I'm supposed to assign it to `state->wnext`.  I wonder why this was never an issue before.

Reply:
Because pointers are passed by reference, not by value. In places where the return value is not ignored, it equals to the final value of first parameter of the function and the result is non-ambiguous.

Reply:
I'm pretty sure pointer arithmetic done in the function will not propagate out the caller unless it was passed as a uint8_t**

Reply:
You have to declare it const to pass by value.

Reply:
I'm pretty sure that const is nothing more than a safeguard for when you derefererence the pointer.  The compiler does not let you modify a parameter like that.  Everything in C is passed by value unless you can take the address of it.  For instance:

```
#include <stdio.h>
#include <stdlib.h>

void some_func(int *in)
{
    in += 5;
}

int main(void)
{
    int *myPtr = NULL;
    some_func(myPtr);
    printf("%p\n", myPtr);

    return 0;
}
```

```[adam@thinkpad ~]$ ./test_ptr.out 
(nil)
```

Reply:
Obviously pointer arithmetic doesn't work with ´NULL` as it's C++ concept and invalid pointer. You can't take address of a pointer in C as it's not guaranteed that the pointer is stored in memory, it can be stored in register and registers don't have an address.

Reply:
If you're referring to the calling convention using registers to pass arguments, the compiler produces epilogues per function such that where the code is called on the stack, all of the register state is preserved. Even when a function gets inlined, code that modified a parameter still preserves the behavior of a function call as the compiler produces a copy if the code modifies a function parameter.

NULL as a pointer doesn't matter in this context, I just chose it as it was something easy to tell if it were modified. It wouldn't matter if the pointer were valid or not. You can try it on a random stack variable's address, as long as it's not dereferenced it will not change the address itself from outside the caller.